scripts/common.make

```diff
@@ -31,6 +31,27 @@ ifdef WINDOWS
 WINDOWS_BUILD=1
 endif
 
+## Usage: $(call remove_unset_options,PREPROCESSOR_INPUT)
+## Remove the preprocessor symbols that are not set in the current configuration
+## from PREPROCESSOR_INPUT. Also normalize whitespace.
+## Example:
+##   $(call remove_unset_options,MBEDTLS_FOO MBEDTLS_BAR)
+## This expands to an empty string "" if MBEDTLS_FOO and MBEDTLS_BAR are both
+## disabled, to "MBEDTLS_FOO" if MBEDTLS_BAR is enabled but MBEDTLS_FOO is
+## disabled, etc.
+##
+## This only works with a Unix-like shell environment (Bourne/POSIX-style shell
+## and standard commands) and a Unix-like compiler (supporting -E). In
+## other environments, the output is likely to be empty.
+define remove_unset_options
+$(strip $(shell
```

Comment:
I thought I know what `strip` does, but what I was thinking of is `tr`. The only `strip` I could find drops symbols from object files. But here in my reading the input to this function is a single line of preprocessor output.

I am a bit confused, could you please explain what is going on here?

Reply:
https://www.gnu.org/software/make/manual/html_node/Text-Functions.html#index-stripping-whitespace

Reply:
[`strip`](https://www.gnu.org/software/make/manual/html_node/Text-Functions.html#index-strip) normalizes whitespace between words and removes leading and trailing whitespace. I guess it gets its name from the second part, which is called `strip` in many languages including Python.